### DISCUSSION TITLE
not detecting wireless adapter

### DISCUSSION THREAD

**Discussion by BDhackers009** [dorssel/usbipd-win#357] [Q&A] [answered]
Hello sir, I wanted to learn something. I just attached my wireless adapter using usbipd-win to WSL(Kali). Now when I type lsusb it is showing the adapter but it didnt showing the adapter in the output of the ifconfig.I tried to run wifite(wireless pentesting tools) but its also not detecting the adapter. I am totally failed to find any solution of that please help me if you can. I am currently running windows 10.

![image](https://user-images.githubusercontent.com/67186139/164950160-aa41ed03-052d-4fc2-ae46-3919d5fcc04e.png)
![image](https://user-images.githubusercontent.com/67186139/164950170-46db1c47-789c-453b-8444-cac53a876852.png)
![image](https://user-images.githubusercontent.com/67186139/164950183-d724cbf7-7577-4bb8-aec5-008f927090ba.png)

**Comment by dorssel** [comment]
@BDhackers009 
This looks like a missing driver in the Linux kernel. The default WSL kernel does not support USB network devices. Did you compile your own? See https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel.

You can look through the (open and closed) issues list for others that had the same experience with USB storage devices, bluetooth, etc. The default WSL kernel only has support for (some) USB-to-serial devices. For anything else, you really need to compile your own.

**Comment by BDhackers009** [comment]
Sir I tried to compile the kernel. Everything was fine but I couldn't find
the bz-Image in arch/x86.....

**Reply by dorssel** [reply]
In the instructions, the `sudo make install` part generates the `bz-Image` at the correct location. Are you sure that step was successful?
Also, someone else already pointed out that after the `cd tools/usb/usbip` part, the command to go back to the original directory is missing. So, the `arch/x86/boot` dir is relative to the `WSL2-Linux-Kernel` dir.

**Reply by BDhackers009** [reply]
Sir I successfully did all things even attached the device into wsl still its detecting the wireless adapter in lsusb but not detecting in ifconfig and airmon-ng

![image](https://user-images.githubusercontent.com/67186139/164970490-3cf809c7-4205-4d72-8489-a43d473f9423.png)

![image](https://user-images.githubusercontent.com/67186139/164970533-85bde549-542f-4be8-a48f-c90f3fe2aeaa.png)

![image](https://user-images.githubusercontent.com/67186139/164970545-ccc0445a-053b-4996-bc5c-0480b8980d00.png)

**Reply by dorssel** [reply]
@BDhackers009 
1) Can you confirm (maybe with `uname -a`) that your private kernel is actually being used?
2) Did you compile your drivers as modules? Maybe you need a `modprobe <module>` to load them?

**Reply by BDhackers009** [reply]
> 1. confirm (m

Here is the output of ` uname -a` 

![image](https://user-images.githubusercontent.com/67186139/164973535-e6d84c43-2e4e-47c8-83f4-b76de7b02843.png)


and sorry I am not a kernel expert I dont know how to use modprobe or what module should I load..can you help me with that sir

**Reply by dorssel** [reply]
From https://linux-hardware.org/?id=usb:148f-7601
Did you configure those CONFIG_XXX ?  If so, did you configure them into the kernel statically (Y) or as a module (M) when you did `make menuconfig`?  If as a module (M), then you need to load them with `modprobe <modulename>`. I don't have this device myself, so I don't know what the module(s) are actually called. All modules are under the `/lib/modules` directory. Just look in there to find if anything looks like it has the right name for this device.

**Reply by BDhackers009** [reply]
SIr I just followed https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel
 this instruction from first to last and I didnt do anything else as you said. I just added CONFIG_USB=y and CONFIG_DEBUG_INFO_BTF=n (it is because the compilation showed an error messege that telling to disable that CONFIG_DEBUG_INFO_BTF)

**Reply by dorssel** [reply]
You will have to be a little bit more creative than that. Of course, you will have to enable exactly the drivers for *your* device. I don't have your device, so I can only point you in the right direction. Now it is up to you.

**Reply by BDhackers009** [reply]
I got it....i am not the right person to do this thing I need to study more about kernel compiling thanks and sorry for wasting your valuable time on me sir.

**Comment by BDhackers009** [comment, marked as answer]
I found that the utility named 'bc' wasn't installed I installed that and
now it's compiling...You should add bc in the list of necessary packages
installation command.

**Reply by dorssel** [reply]
@benmcmorran 
NOTE: another package missing from compile instructions

**Reply by marco73** [reply]
When building for a new kali-linux WSL these where all packages I needed to install:
apt install git usbutils make libncurses-dev gcc bison flex dwarves libssl-dev libelf-dev python3 bc usbip
